backend/assignments/2/src/main/java/com/example/demo/controller/GeocodingController.java

```diff
@@ -0,0 +1,32 @@
+package com.example.demo.controller;
+import com.example.demo.dto.AddressDto;
+import com.example.demo.dto.CoordinatesDto;
+import com.example.demo.entity.Address;
+import com.example.demo.entity.Coordinates;
+import com.example.demo.service.GeocodingService;
+import com.example.demo.service.ReverseService;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@AllArgsConstructor
+public class GeocodingController {
+
+    private GeocodingService geocodingService;
+    private ReverseService reverseService;
+    @GetMapping("/geocoding")
+    public ResponseEntity<Coordinates> getCoordinates(@RequestParam(required = true, name = "address") AddressDto addressDto){
```

Comment:
FYI
For security reasons, we never expose the entity in response.
Use a seperate response DTO for the same